src/Either.js

```diff
@@ -1,61 +1,69 @@
-module.exports = Either;
+var util = require('./internal/util');
 
-function F() {}
+function Either() {
+    throw new Error('Either is not directly callable');
+}
 
-function inherit(Parent, Child) {
-    if (Object.create) {
-        Child.prototype = Object.create(Parent.prototype);
-    } else {
-        F.prototype = Parent.prototype;
-        Child.prototype = new F();
-        Child.prototype.constructor = Child;
-    }
-    return Child;
+Either.prototype.map = util.returnThis;
+
+Either.of = Either.prototype.of = function(value) { 
+    return Either.Right(value); 
+};
+
+Either.prototype.chain = util.returnThis; // throw?
+
+Either.equals = Either.prototype.equals = function(that) {
+    return this.constructor === that.constructor && this.value === that.value;
+};
+
+
+// Right
+function _Right(x) {
+    this.value = x;
 }
+util.extend(_Right, Either);
+
+_Right.prototype.map = function(fn) { 
+    return new _Right(fn(this.value)); 
+};
+
+_Right.prototype.ap = function(that) { 
+    return that.map(this.value); 
+};
+
+_Right.prototype.chain = function(f) { 
+    return f(this.value); 
+};
+
+Either.Right = function(value) {
+    return new _Right(value);
+};
+
 
-function Either(left, right) {
+// Left
+function _Left(x) {
+    this.value = x;
+}
+util.extend(_Left, Either);
+
+_Left.prototype.ap = function(that) { return that; };
+
+Either.Left = function(value) {
+    return new _Left(value);
+};
+
+
+Either.fromNullable = function(left, right) {
```

Comment:
Shouldn't this be curried? 
And if it is, do we decide to depend on `ramda` here? Perhaps it makes sense to have `ramda` as a dependency in `ramda-fantasy`